---
title: "Hypertension PRS"
author: "Nuzulul Kurniansyah & Tamar Sofer"
date: "9/20/2021"
output: md_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

This is the instruction to construct, generate and performs the association test for Polygenic Risk Score in Hypertension. This instruction is based on the manuscript A multi-ethnic polygenic risk score is associated with hypertension prevalence and progression throughout adulthood.


## STEP 1: Installation and require R packages

In this paper, We used [PRSice 2.3.1.e](https://www.prsice.info "PRSice 2.3.1.e") to generate PRS. 

We performed the analysis using R/4.0.2 programming. We first need to install the required CRAN (dplyr, tidyverse, data.table, purrr, pROC) and BioConductor(GENESIS, GWASToools) packages

```{r eval=FALSE, echo=TRUE}

install.packages("dplyr")

```


## STEP 2: Construct PRS

We used hypertension “pan ancestry” GWAS from [UKBB](https://pan.ukbb.broadinstitute.org), and systolic BP ([SBP](https://pubmed.ncbi.nlm.nih.gov/30578418/)), and diastolic BP ([DBP](https://pubmed.ncbi.nlm.nih.gov/30578418/)) from MVP(Million Veteran Program) to construct PRS (See script below). Then we applied CV (Coefecient Variation) to select best perfomance PRS, see manuscript for more detail. 

We provide summary statistics to crete HTN PRS in this repistory(./Summary_Statitcs/*). The summary statitics created are cretaed based on clumping paramenter below:

```{r}
pan_ancestry<- data.frame(Threshold="0.2", Distance="250kb", R2="0.1",Trait="HTN", Study="Pan-UKBB")
mvp_DBP<- data.frame(Threshold="0.1", Distance="1000kb", R2="0.1", Trait="DBP",Study="MVP")
mvp_SBP<- data.frame(Threshold="0.01", Distance="1000kb", R2="0.1", Trait="SBP",Study="MVP")

res<-rbind(pan_ancestry,mvp_DBP,mvp_DBP)

kableExtra::kable(res, caption = "Clumping paramaters based on CV approach")

```



```{bash eval=FALSE, echo=TRUE}


Rscript ./PRSice.R \
 --dir ./PRS_Output \
 --prsice ./PRSice_linux/PRSice_linux \
 --base ./Summary_Statistic/. \
 --target ./Genotype \
 --thread 2 \
 --chr Chromosome 
 --bp Position 
 --A1 Allele1 
 --A2 Allele2 
 --pvalue PValue \
 --bar-levels PValue \
 --stat BETA 
 --all-score T \
 --out ./out_prs \
 --no-clump T
 --print-snp T \
 --ignore-fid T 
 --no-regress T 
 --fastscore T 
 --model add 
 --no-full T 
 --chr-id c:L:a:B 


```



## STEP 3: Construct PRSsum

After run PRS for each summary statistics, then we combine PRS (PRSsum).

```{r eval=FALSE, echo=TRUE}
library(data.table)
library(dplyr)
library(purrr)


prs_trait<- c("SBP", "DBP","HTN")
out<-list()
for(prs in prs_trait){
  
  
  prs_output<-paste0("./", prs,".txt")
  prs_df<-fread(prs_output, data.table=F)
  prs_df<- prs_df %>% dplyr::select(-IID)
  colnames(prs_df)<- c("sample.id",prs)
  
  #standardize prs

  prs_df[, prs]<- (prs_df[,prs] - mean(prs_df[,prs]))/sd(prs_df[,prs])
  out[[prs]]<- prs_df
  
  
}

combine_prs<- purrr::reduce(out, full_join , by="sample.id")



prssum<- data.frame(sample.id=prssum$sample.id, PRSsum=apply(prssum[,], 1, sum))

prssum[,"PRSsum"]<- (prssum[,"PRSsum"] - mean(prssum[,"PRSsum"]))/sd(prssum[,"PRSsum"])



```

## STEP 3: Perform Association Analysis

Finally, we perormed association analysis using mix model.

```{r eval=FALSE, echo=TRUE}

library(GENESIS)
library(GWASTools)
library(pROC)


source("./Code/Utils.R")


#phenotype

pheno<- fread(phenotype_file, data.table=F)


# merge PRSsum with phenotype

pheno_df<-left_join(pheno,prssum, by="sample.id" )


covarites_prs<- c("BMI","age","sex","site","race",paste0("PC_",1:11),"PRSsum")

outcome<-"HTN"

## Kinship matrix

covMatlist<-getobj(covMatlist)


assoc_df<- run_assoc_mixmodel(pheno=pheno,
                              outcome=outcome,
                              covars_prs=covarites_prs, 
                              covmat=covMatlist,
                              group.var=NULL)


# Perform AUC

#only use unrelated people

unrels<- getobj(unrels_people)

pheno_unrels<- pheno[pheno_df$sample.id %in% unrels,]


auc<- generate_auc(pheno=pheno_unrels,
                   outcome=outcome,
                   covars_prs=covarites_prs, seed=NULL,
                   n= 2000)



final_assoc<-c(assoc_df,auc)


```











