---
title: "Hypertension PRS"
author: "Nuzulul Kurniansyah & Tamar Sofer"
date: "10/08/2021"
output: md_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

This repository provides information regarding the construction of a polygenic risk score (PRS) for hypertension (HTN) that we developed (HTN-PRS) and will be posted as a preprint soon (link to be added).

First, it provides instructions for constructing the HTN-PRS based on summary statistics from GWAS. We provide the relevant summary statistics (see folder "Summary_Statistics_for_PRS_construction"), as well as code for using them to construct the PRS.  

Second, this repository also provides code the we used for the analyses in the manuscript (see folder "Code"). 


## Required packages

We used [PRSice 2.3.1.e](https://www.prsice.info "PRSice 2.3.1.e") to generate PRS. We provide example code that also uses PRSice to construct PRS based on the provided summary statistics in folder "Summary_Statistics_for_PRS_construction".

Other software and packages that we used, but may not be necessary for others to construct the PRS, are as follows: 
1. We performed the analysis using R version 4.0.2. W
2. We used the following packages from CRAN: dplyr, tidyverse, data.table, purrr, pROC.
3. We used the following packages from BioConductor: GENESIS, GWASTools.

```{r eval=FALSE, echo=TRUE}

install.packages("dplyr")

```


## PRS construction
Our HTN-PRS is a sum of multiple trait-specific PRS (HTN, systolic blood pressure (SBP) and diastolic blood pressure (DBP)). Summary statistics to create the each of the trait-PRS are provided here in a subfolder(./Summary_Statistics_for_PRS_construction/*). 

Specific GWAS used: hypertension “pan ancestry” GWAS from [UKBB](https://pan.ukbb.broadinstitute.org),   ([SBP GWAS from MVP](https://pubmed.ncbi.nlm.nih.gov/30578418/)), and ([DBP GWAS from MVP](https://pubmed.ncbi.nlm.nih.gov/30578418/)) with MVP standing for Million Veteran Program. 

The summary statistics provided in this repository were selected from those in the complete GWAS based on clumping parameter below, where we used the multi-ethnic TOPMed dataset used in the paper as an LD reference panel. To select the specific tuning parameter (LD parameters, p-value threshold) for each trait-specific PRS, we applied an approach where we optimized the coefficient of variation (CV) computed over the estimated effect sizes of the candidate PRS across 5 independent subset of the training dataset. See manuscript for more detail. 


```{r echo=FALSE}
pan_ancestry<- data.frame(Threshold="0.2", Distance="250kb", R2="0.1",Trait="HTN", Study="Pan-UKBB", "Topmed mean"=4.46e-06, "Topmed sd"=4.07e-06)
mvp_DBP<- data.frame(Threshold="0.1", Distance="1000kb", R2="0.1", Trait="DBP",Study="MVP","Topmed mean"=-4.95e-04, "Topmed sd"=2.63e-04)
mvp_SBP<- data.frame(Threshold="0.01", Distance="1000kb", R2="0.1", Trait="SBP",Study="MVP","Topmed mean"=-4.63e-03, "Topmed sd"=1.11e-03)

res<-rbind(pan_ancestry,mvp_DBP,mvp_DBP)

kableExtra::kable(res)

```




```{bash eval=FALSE, echo=TRUE}


Rscript ./PRSice.R \
 --dir ./PRS_Output \
 --prsice ./PRSice_linux/PRSice_linux \
 --base ./Summary_Statistic/. \
 --target ./Genotype \
 --thread 2 \
 --chr Chromosome 
 --bp Position 
 --A1 Allele1 
 --A2 Allele2 
 --pvalue PValue \
 --bar-levels Threshold \
 --stat BETA 
 --all-score T \
 --out ./out_prs \
 --no-clump T
 --print-snp T \
 --ignore-fid T 
 --no-regress T 
 --fastscore T 
 --model add 
 --no-full T 
 --chr-id c:l:a:b


```



## STEP 3: Construct PRSsum

After run PRS for each summary statistics, then we combine PRS (PRSsum).


```{r}


Topmed_stage2<- data.frame("TOPMed mean"="-5.86e-16", "TOPMed sd"= "2.31" )

kableExtra::kable(Topmed_stage2, caption = "Mean and SD to stndardize PRSsum")
```


```{r eval=FALSE, echo=TRUE}
library(data.table)
library(dplyr)
library(purrr)


prs_trait<- c("SBP", "DBP","HTN")
out<-list()
for(prs in prs_trait){
  
  
  prs_output<-paste0("./", prs,".txt")
  prs_df<-fread(prs_output, data.table=F)
  prs_df<- prs_df %>% dplyr::select(-IID)
  colnames(prs_df)<- c("sample.id",prs)
  
  #standardize prs, Use the mean and sd PRS from TOPMed 

  prs_df[, prs]<- (prs_df[,prs] - mean(prs_df[,prs]))/sd(prs_df[,prs])
  out[[prs]]<- prs_df
  
  
}

combine_prs<- purrr::reduce(out, full_join , by="sample.id")



prssum<- data.frame(sample.id=prssum$sample.id, PRSsum=apply(prssum[,], 1, sum))

prssum[,"PRSsum"]<- (prssum[,"PRSsum"] - mean(prssum[,"PRSsum"]))/sd(prssum[,"PRSsum"])



```

## STEP 3: Perform Association Analysis

Finally, we perormed association analysis using mix model.

```{r eval=FALSE, echo=TRUE}

library(GENESIS)
library(GWASTools)
library(pROC)


source("./Code/Utils.R")


#phenotype

pheno<- fread(phenotype_file, data.table=F)


# merge PRSsum with phenotype

pheno_df<-left_join(pheno,prssum, by="sample.id" )


covarites_prs<- c("BMI","age","sex","site","race",paste0("PC_",1:11),"PRSsum")

outcome<-"HTN"

## Kinship matrix

covMatlist<-getobj(covMatlist)


assoc_df<- run_assoc_mixmodel(pheno=pheno,
                              outcome=outcome,
                              covars_prs=covarites_prs, 
                              covmat=covMatlist,
                              group.var=NULL)


# Perform AUC

#only use unrelated people

unrels<- getobj(unrels_people)

pheno_unrels<- pheno[pheno_df$sample.id %in% unrels,]


auc<- generate_auc(pheno=pheno_unrels,
                   outcome=outcome,
                   covars_prs=covarites_prs, seed=NULL,
                   n= 2000)



final_assoc<-c(assoc_df,auc)


```











